### ISSUE TITLE
[BUG] UWUFETCH_VERSION string not set in Github generated archives

### ISSUE THREAD

**Issue by antecrescent** [TheDarkBug/uwufetch#247]
**Describe the bug**
Github automatically creates [source code archives](https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives#source-code-archive-urls) which are commonly used by distros to fetch repositories at point releases.
These do not include the `.git` directory. Consequently, `UWUFETCH_VERSION = $(shell git describe --tags)` sets an empty string at compile time.

**To Reproduce**
1. Download https://github.com/github/TheDarkBug/uwufetch/refs/heads/main.tar.gz
2. Run `make build`
3. See described result in e.g. the cc flags.

**Expected behavior**
Sensible version string regardless of vcs presence in archive. I propose to hardcode the string and update it upon release of a new version. Another option would be to allow users to pass the string as a make argument but this doesn't seem correct to me.

**Desktop (please complete the following information):**
 - OS: any
 - Version since 2.0

**Are you already working on this bug?**
I want your feedback on my proposition before proceeding with my proposition.

**Comment by TheDarkBug** [comment]
Yes this is an issue I didn't think about, what solution would you propose?

**Comment by antecrescent** [comment]
Since `UWUFETCH_VERSION` only really has meaning at the release versions, I suggest hard-coding and incrementing it upon release.

**Comment by TheDarkBug** [comment]
You are right, maybe a better solution would be to add compiler version and other compile-time info for debugging

**Comment by TheDarkBug** [comment]
I implemented a more complex way to "hard code" version tags. Now compiler info is included (for possible debugging purposes), git tag description and the version tag (hard coded as you suggested). Of course the bug will be actually fixed in the next release. I will close the issue before releasing 2.2 (or maybe 3.0, idk)
